src/main/java/dev/meyi/bn/utilities/EnchantedCraftingHandler.java

```diff
@@ -0,0 +1,155 @@
+                if(h % 2 == 0){
+                    IngredientP = BazaarNotifier.bazaarDataRaw.getJSONObject(BazaarNotifier.enchantCraftingList.getJSONObject("Other").getJSONObject(ItemName).getJSONArray("Material").getString(h)).getJSONArray("sell_summary").getJSONObject(0).getDouble("pricePerUnit");
+                }else{
+                    IngredientC = BazaarNotifier.enchantCraftingList.getJSONObject("Other").getJSONObject(ItemName).getJSONArray("Material").getInt(h);
+                    P += (IngredientP*IngredientC);
+                }
+            }
+
+
+
+            Double Price1 = ItemSellPrice - P;
+            double Price2 = ItemBuyPrice - P;
+            Double CountPerMil = 1000000 / P;
+            double PricePerMil = CountPerMil * Price1;
+            list.add(new ArrayList<>(Arrays.asList(Price1.toString(), Double.toString(Price2), Double.toString(PricePerMil), ItemName)));
+
+        }
+
+
+
+
+        list.sort((o1, o2) -> {
+            ArrayList<Double> list1 = new ArrayList<>();
+            ArrayList<Double> list2 = new ArrayList<>();
+            list1.add(Double.valueOf(o1.get(0)));
+            list1.add(Double.valueOf(o1.get(1)));
+            list1.add(Double.valueOf(o1.get(2)));
+            list2.add(Double.valueOf(o2.get(0)));
+            list2.add(Double.valueOf(o2.get(1)));
+            list2.add(Double.valueOf(o2.get(2)));
+
+
+            return list1.get(BazaarNotifier.config.getInt("CraftingGUITOGGLE")).compareTo(list2.get(BazaarNotifier.config.getInt("CraftingGUITOGGLE")));
+        });
+        Collections.reverse(list);
+        return list;
+    }
+
+
+    public static String toggleCrafting(){
+        if(BazaarNotifier.config.getInt("CraftingGUITOGGLE")==0){
+            BazaarNotifier.config.put("CraftingGUITOGGLE", 1);
+            return "Toggled crafting to selloffer";
+        }else if(BazaarNotifier.config.getInt("CraftingGUITOGGLE")==1){
+            BazaarNotifier.config.put("CraftingGUITOGGLE", 2);
+            return "Toggled crafting to profit per million";
+        }else {
+            BazaarNotifier.config.put("CraftingGUITOGGLE", 0);
+            return "Toggled crafting to instasell";
+        }
+    }
+
+    public static String setCraftingLength(int Length){
+        if(Length >= BazaarNotifier.enchantCraftingList.getJSONObject("Normal").length()+BazaarNotifier.enchantCraftingList.getJSONObject("Other").length()-1){
+            return "Error Number to high";
+        }else{
+            BazaarNotifier.config.put("CraftingListLength" , Length);
+            return "Setting CraftingListLength to " + Length;
+        }
+    }
+
+
+    public static String[] getEnchantCraft(String itemU){
+        String ItemName;
+        try{ItemName = BazaarNotifier.bazaarConversionsReversed.getString(WordUtils.capitalize(itemU.toLowerCase()));}catch (Exception e){ItemName = "";}
+        String[] values = new String[3];
+        try {

```

Comment:
Can we get rid of this try-catch? (same reason as mentioned earlier)

Reply:
probably Yes